streamPair.q

```diff
@@ -0,0 +1,80 @@
+// load tables
+tab1: 40000#1_ flip `dateTime`bid`ask`bidVol`askVol!("*FFFF";",") 0: `:C:/q/dash/sample/data/stocks/USA500IDXUSD.csv;
+tab2: 40000#1_ flip `dateTime`bid`ask`bidVol`askVol!("*FFFF";",") 0: `:C:/q/dash/sample/data/stocks/USATECHIDXUSD.csv;
+//tab1: 1_ flip `dateTime`bid`ask`bidVol`askVol`Vol!("DFFFFF";",") 0: `:C:/q/dash/sample/data/stocks/EWA2.csv;
+//tab2: 1_ flip `dateTime`bid`ask`bidVol`askVol`Vol!("DFFFFF";",") 0: `:C:/q/dash/sample/data/stocks/EWC2.csv;
+tab3: flip `dateTime`spread`mean`up`low`operation!("P"$();"F"$();"F"$();"F"$();"F"$();"F"$());
+
+
+// Fix Time
+update dateTime:"P"$@[;19;:;"."] each dateTime from `tab1;
+update dateTime:"P"$@[;19;:;"."] each dateTime from `tab2;
+
+// Take date and log(prices)
+priceX: distinct select distinct dateTime, log bid, log ask from tab1;
+priceY: distinct select distinct dateTime, log bid, log ask from tab2;
+spreads: 200#select from tab3;
+
+// Eliminate nulls and filter data
+update delta:0f^deltas dateTime from `priceX;
+update delta:0f^deltas dateTime from `priceY;
+priceX: 0!1_priceX;
+priceY: 0!1_priceY;
+
+// Create an empty auxiliary table
+tAux: 1_1#priceX;
+profit: 0;
+
+// Alpha and beta functions
+betaF:{dot:{sum x*y};                                      
+      ((n*dot[x;y])-(*/)(sum')(x;y))%                         
+      ((n:count[x])*dot[x;x])-sum[x]xexp 2};
+alphaF: {avg[y]-(betaF[x;y]*avg[x])};
+
+// Calculate alpha y beta and spreads
+px: exec bid from priceX;
+py: exec bid from priceY;
+s1: py - ((px * beta[px;py])-alpha[px;py]);
+spreads: update dateTime: priceX[`dateTime], spread: s1, mean: avg[100#s1] ,up: avg[100#s1]+1.96*dev[100#s1], low: avg[100#s1]-1.96*dev[100#s1] , operation:0 from spreads;

```

Comment:
Calculate `avg[100#s1]` and `dev[100#s1]` just once

Reply:
okeys